### PR TITLE
Add how to write scalar functions

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://databend.rs/policies/cla/

## Summary

Summary about this PR

## Changelog

 - Documentation
 

## Related Issues

  #2563

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

**Comment by sundy-li** [comment]
cc @Junnplus  @mshauneu, may be interested in reviewing this.

File: website/databend/docs/development/how-to-write-scalar-functions.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+#### Constant Column case
+Noticed that we should take care of the Constant Column match case to improve memory usage.
+
+
+#### Column Iteration and validity bitmap combination
+To iterate the column, we can use `column.iter()` to generate an iterator, the item is `Option<T>`, `None` means it's null.
+But this's inefficient because we need to check the null value every time we iterate the column inside the loop which will pollute the CPU cache.
+
+According to the memory layout of Arrow, we can directly use the validity bitmap of the original column to indicate the null value.
+So we have `ArrayApply` trait to help you iterate the column. If there are two zip iterator, we can use `binary` function to combine the validity bitmap of two columns.
```
Comment (Junnplus) [inline website/databend/docs/development/how-to-write-scalar-functions.md:143]: Good job!

BTW, show code may be easier to get started.

**Comment by BohuTANG** [comment]
Great job, nice guideline~

**Pull request merged**